swift/Tasks/DittoManager.swift

```diff
@@ -2,41 +2,368 @@ import DittoSwift
+    ///
+    /// This method sets up a sync subscription that:
+    /// - Creates a subscription for the entire tasks collection
+    /// - Enables data to be synced between devices when sync is active
+    /// - Stores the subscription reference for later cancellation
+    ///
+    /// The subscription remains active until explicitly cancelled or when sync is stopped.
+    ///
+    /// - SeeAlso: https://docs.ditto.live/sdk/latest/sync/syncing-data#creating-subscriptions
+    ///
+    /// - Throws: A DittoError if the subscription registration fails
+    func registerSubscription() throws {
+        if let ditto {
+            // https://docs.ditto.live/sdk/latest/sync/syncing-data#creating-subscriptions
+            let subscriptionQuery = "SELECT * from tasks"
+            subscription = try ditto.sync.registerSubscription(
+                query: subscriptionQuery
+            )
+        }
+    }
+
+    /// Enables or disables Ditto sync based on the provided value.
+    ///
+    /// This method checks the current sync state and starts or stops
+    /// Ditto sync accordingly:
+    /// - If `newValue` is `true` and sync is not active, it starts sync.
+    /// - If `newValue` is `false` and sync is active, it stops sync.
+    ///
+    /// - Parameter newValue: A Boolean indicating whether sync should be enabled (`true`) or disabled (`false`).
+    /// - Throws: A DittoError if the operation fails
+    func setSyncEnabled(_ newValue: Bool) throws {
+        if let ditto {
+            if !ditto.isSyncActive && newValue {
+                try startSync()
+            } else if ditto.isSyncActive && !newValue {
+                stopSync()
+            }
+        }
+    }
+
+    /// Starts Ditto sync and registers a subscription for the tasks collection.
+    ///
+    /// This method:
+    /// - Initiates the Ditto sync process if the Ditto instance is available.
+    ///
+    /// - SeeAlso: https://docs.ditto.live/sdk/latest/install-guides/swift#integrating-and-initializing-sync
+    /// - Throws: A DittoError if the operation fails
+    private func startSync() throws {
+        if let ditto {
+            // https://docs.ditto.live/sdk/latest/install-guides/swift#integrating-and-initializing-sync
+            try ditto.startSync()
+        }
+    }
+
+    /// Stops Ditto sync and cancels the active subscription for the tasks collection.
+    ///
+    /// This method:
+    /// - Cancels the current sync subscription to stop receiving updates from remote peers.
+    /// - Sets the subscription property to `nil`.
+    /// - Stops the Ditto sync process if it is active.
+    ///
+    /// - SeeAlso: https://docs.ditto.live/sdk/latest/sync/syncing-data#canceling-subscriptions
+    /// - Throws: A DittoError if the operation fails
+    private func stopSync() {
+        // https://docs.ditto.live/sdk/latest/sync/syncing-data#canceling-subscriptions
+        subscription?.cancel()
+        subscription = nil
+
+        ditto?.stopSync()
+    }
```

Comment:
Looks like registration has been removed from `startSync()` but cancelling has been left in here.

```suggestion
    private func stopSync() {
        ditto?.stopSync()
    }
```

Also, given that start/stopSync() are just calling through, maybe get rid of those wrappers entirely and call directly the ones on Ditto?

Reply:
This isn't a real app - it's a place someone can look to see an implementation of Ditto.  If it was a real app, I would have written this a lot different.   Removing the sync stop and stop functions defeats the purpose of the Manager class.  The wrapper provides:
- Abstraction from implementation - good for testing
- Everything that talks to Ditto is in one spot.  Having to jump all over the place to figure out how something works isn't fun when you are learning.   I know from countless conversations with developers.   It's why I refactored into the Manager class in the first place to make it so everything you need to learn or see is in one place.

The added advantage is you can copy and paste a single file into any starter app and have something up and running pretty quickly when you want to make a new POC.  At the previous company we did that a lot with our "learning path" app that we created that serves a very similar purpose.